doc: Added current results demo and reproduction process markdown (#26)
pilagod/waallet-extension: demo.md

# Results Demonstration and Reproduction Process

## Demo 1: Swap ETH for UNI tokens in the AA contract on the Sepolia testnet within the Uniswap DApp.

[![Waallet Demo](https://i.imgur.com/f1QHreR.gif)](https://youtu.be/PviRgMoooII)

### Reproduce the process

#### (1) Deploy your AA SimpleAccount contract

- Deploy this contract to Sepolia testnet in [Remix](https://remix.ethereum.org/) and execute the `createAccount(address owner, uint256 salt)` function.
- Note the `NewAccount` event to find the address of the deployed AA SimpleAccount contract.

> - Also transfer 0.1 SepoliaETH to the SimpleAccount contract.

Reviewer: ```suggestion
- Remember to transfer some SepoliaETH (0.1 or so) to the SimpleAccount contract.
```

> ```typescript

Reviewer: ```suggestion
```solidity
```

// SPDX-License-Identifier: UNLICENSED

> pragma solidity >=0.8.2 <0.9.0;

Reviewer: ```suggestion
pragma solidity ^0.8.0;
```

contract DeployAccount {
    IEntryPoint entryPoint = IEntryPoint(0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789);
    SimpleAccountFactory accountFactory = new SimpleAccountFactory(entryPoint);

    event NewAccount(address indexed account, address indexed accountFactory);

    function createAccount(address owner,uint256 salt) public {
        SimpleAccount account = accountFactory.createAccount(owner, salt);
        emit NewAccount(address(account), address(accountFactory));
    }

    function getAddress(address owner,uint256 salt) public view returns (address) {
        return accountFactory.getAddress(owner, salt);
    }
}
```

#### (2) Download and compile the waallet-extension project

- Download the waallet-extension project and its dependencies.

```shell
### Download the waallet-extension project and its dependencies.
git clone git@github.com:pilagod/waallet-extension.git
cd waallet-extension/
git switch --detach uniswap-swap
npm install

### Edit the .env file's environment variables.
cp .env.example .env
vim .env
```

#### (3) Edit the .env file's environment variables.

```shell

> PLASMO_PUBLIC_SHIP_NAME=ncc-1701
> PLASMO_PUBLIC_SHIELD_FREQUENCY=147
> PLASMO_PUBLIC_SITE_URL=https://www.plasmo.com
> 
> PLASMO_PUBLIC_ACCOUNT=<YOUR_SIMPLE_ACCOUNT_ADDRESS>
> PLASMO_PUBLIC_ACCOUNT_OWNER_PRIVATE_KEY=<YOUR_SIMPLE_ACCOUNT_OWNER_PRIVATE_KEY>
> PLASMO_PUBLIC_NODE_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/<YOUR_ALCHEMY_KEY>
> PLASMO_PUBLIC_BUNDLER_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/<YOUR_ALCHEMY_KEY>
> 
> CODE=PLASMO-GO-030 # Undefined in extension because it does not start with PLASMO_PUBLIC_
> ```

Reviewer: ```suggestion
PLASMO_PUBLIC_ACCOUNT=<YOUR_SIMPLE_ACCOUNT_ADDRESS>
PLASMO_PUBLIC_ACCOUNT_OWNER_PRIVATE_KEY=<YOUR_SIMPLE_ACCOUNT_OWNER_PRIVATE_KEY>
PLASMO_PUBLIC_NODE_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/<YOUR_ALCHEMY_KEY>
PLASMO_PUBLIC_BUNDLER_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/<YOUR_ALCHEMY_KEY>
```
